src/components/common/ButtonWithIcon.vue

```diff
@@ -1,27 +1,42 @@
 <template>
   <BaseButton
+    class="flex items-center"
     :type="type"
     :disabled="disabled"
     :size="size"
   >
-    <div class="flex items-center">
-      <slot name="icon" />
-      <slot name="text" />
-    </div>
+    <fa
+      :class="classes"
+      class="mr-3"
+      :size="size"
+      :icon="icon"
+    />
+    <slot />
   </BaseButton>
 </template>
 
 <script>
 import BaseButton from "../../components/common/BaseButton";
+import Fa from 'vue-fa'
+import { faStar, faFlag, faPlus } from '@fortawesome/free-solid-svg-icons'
```

Comment:
Are we only going to be allowing these three icons? The way you are doing it right now requires we import the icons we want to use in with `ButtonWithIcon` component. Meaning, that every time we want to use a new icon we need to import it here. 
I believe the way it was before (importing the icon in the parent and setting it with a slot) makes a lot more sense. Just imagine we use 20+ icons. This file will become a complete mess!

Reply:
the issue that I was facing with the previous method is that i can't, or well, don't know, how to give that icon an approriate class from inside the component, meaning that the color of the icon would not depend on the type prop of the button component.

Reply:
Check code suggestion below. Otherwise here again. The Icon will automatically scale up if it is in a container with a custom font size set.